File: papi/lista2/exercicio3.py
print("Faremos a soma de 2 números")

while True:
    try:
        numero1 = int(input("Por favor informe o primeiro número: "))
        break
    except ValueError:
        print("Por favor informe um número válido.")

while True:
    try:
        numero2 = int(input("Por favor informe o segundo número: "))
        break
    except ValueError:
        print("Por favor informe um número válido.")


print("A soma dos números é", numero1 + numero2)

File: papi/lista2/exercicio3.php
<?php
while (true) {
    $numero1 = (int) readline("Por favor forneça o primeiro número diferente de zero: ");
    if ($numero1 == 0) {
        echo "O valor atribuído é equivalente ao número zero. Por favor, tente novamente.\n";
    } else {
        break;
    }
}

while (true) {
    $numero2 = (int) readline("Por favor forneça o segundo número diferente de zero: ");
    if ($numero2 == 0) {
        echo "O valor atribuído é equivalente ao número zero. Por favor, tente novamente.\n";
    } else {
        break;
    }
}

echo "A soma destes números é: ";
echo $numero1 + $numero2;
echo "\n";